src/Perpetuum/Services/EventServices/EventProcessors/EnvironEffectHandler.cs

```diff
@@ -0,0 +1,150 @@
+﻿using Perpetuum.ExportedTypes;
+using Perpetuum.Services.Daytime;
+using Perpetuum.Services.EventServices.EventMessages;
+using Perpetuum.Services.Weather;
+using Perpetuum.Zones;
+using Perpetuum.Zones.Effects.ZoneEffects;
+using System;
+using System.Collections.Generic;
+
+namespace Perpetuum.Services.EventServices.EventProcessors
+{
+    /// <summary>
+    /// Environment Event processor to modify ZoneEffects on a zone based on weather and time of day
+    /// </summary>
+    public class EnvironEffectHandler : EventProcessor<EventMessage>
+    {
+        private readonly IZone _zone;
+        private WeatherInfo _weatherState;
+        private GameTimeInfo _gameTime;
+        private ZoneEffect _currentEffect;
+        private readonly EffectType[] _effectTypes = new EffectType[] {
+            EffectType.effect_day,
+            EffectType.effect_day_clear,
+            EffectType.effect_day_overcast,
+            EffectType.effect_night,
+            EffectType.effect_night_clear,
+            EffectType.effect_night_overcast,
+            EffectType.effect_weather_good,
+            EffectType.effect_weather_bad
+        };
+        private readonly IDictionary<EffectType, Lazy<ZoneEffect>> _effects = new Dictionary<EffectType, Lazy<ZoneEffect>>();
+        public EnvironEffectHandler(IZone zone)
+        {
+            _zone = zone;
+            _effects = InitEffectCollection(_effectTypes);
+        }
+
+        private IDictionary<EffectType, Lazy<ZoneEffect>> InitEffectCollection(EffectType[] effectTypes)
+        {
+            var dict = new Dictionary<EffectType, Lazy<ZoneEffect>>();
+            foreach (var effType in effectTypes)
+            {
+                dict.Add(effType, new Lazy<ZoneEffect>(() => new ZoneEffect(_zone.Id, effType, true)));
```

Comment:
@DerekJarvis @nref-dan These are lazy because if these are constructed on-init of the zone, it will break because the .Id property isn't resolved at that time.  But probably there is another way to register this into the bootstrapper and avoid this.  So I'll admit its a bit of a hack, and really they dont *need* to be lazy as its really only handling the case of when the constructor of this class is called before the zone is fully initialized.